pkg/skaffold/sync/syncer_mux.go

```diff
@@ -33,13 +35,21 @@ func (s SyncerMux) Sync(ctx context.Context, out io.Writer, item *Item) error {
 		}
 	}
 
-	// Return an error only if all syncers fail
-	if len(errs) == len(s) {
-		var err error
+	if len(errs) > 0 {
+		err := fmt.Errorf("sync failed for artifact %q", item.Image)
```

Comment:
do we need to wrap with this error? I think this might lead to something a bit awkward looking, i.e.
```
sync failed for artifact foo: copying files: sync failed for artifact foo
```
unless I'm misunderstanding the code path. Wdyt @renzodavid9 ?

Reply:
You are right, the message will have all the errors found during sync. In this case, and due to #7447, the message will look something like this:

`Skipping deploy due to sync error:copying files: didn't sync any files: copying files: didn't sync any files: sync failed for artifact foo`

Not sure then if it's better to manage this error here differently (without doing the `for` for the wrapping) or just wait until we solve #7447 to avoid having the same message duplicated in the error.

Reply:
I think maybe we can just leave it for now, and if we need to clean up because it's confusing we can do so later